### PR TITLE
OSAC-4486: Add enclave-cert-gen and enable real-CA e2e on demand

### PR DESCRIPTION
**Pull request by rporres**
Third and final PR in the series addressing gori-project/GoRI#943, which identified that all test certificates used a custom test CA and scenario 1 (well-known / system CA) was never exercised in CI. PR #609 added cert helpers and multi-hop chain / expiry test coverage; PR #613 moved all cert validation into the Python CLI and removed the bash cert functions. This PR closes the remaining gap by issuing real certificates from public CAs and using them in the disconnected e2e.

Introduces enclave-cert-gen, a Click CLI tool that issues certificates via certbot with Hetzner DNS-01 validation against Let's Encrypt and ZeroSSL, rotating through five CA types by day of week on the nightly schedule. The logic is embedded directly in the existing generate_enclave_vars.sh and generate_ironic_cert.sh scripts: when ENCLAVE_CERT_TYPE is set they call the tool; otherwise the self-signed path runs unchanged.

Real certs are also available on demand: a cert-type input on both workflow_dispatch and workflow_call accepts a specific type or "auto" (day-of-week rotation), defaulting to self-signed when omitted. The cert resolution step is unconditional and writes both ENCLAVE_CERT_TYPE and ENCLAVE_BASE_DOMAIN so they always agree, regardless of trigger. The dead /test check-real-certs slash command, which pointed at a workflow removed in PR #613, is also cleaned up.

## Summary by CodeRabbit

* **New Features**
  * Added automated certificate generation for enclave deployments.
  * Added Let’s Encrypt and ZeroSSL support with RSA and ECDSA options.
  * Added configurable certificate types for deployment workflows.
  * Added custom base domain support, defaulting to `lab`.
  * Added certificate output for ingress API and Ironic services.

* **Bug Fixes**
  * Improved certificate handling, deployment diagnostics, and DNS validation messages.
  * Improved deployment setup reliability by ensuring required tooling is available.
  * Improved Clair validation retries and failure diagnostics.

### CODE REVIEW

File: .github/workflows/e2e-deployment.yml [hunk 1]
```diff
@@ -72,6 +72,19 @@ on:
...
+        type: choice
+        default: ''
+        options:
+          - ''
+          - auto
+          - le-rsa
+          - le-ecdsa-long
+          - le-ecdsa-short
+          - zerossl-rsa
+          - zerossl-ecdsa
```
[the remark this answers is not in the document]
Comment (rporres) [reply .github/workflows/e2e-deployment.yml:87]: ✨ **Claude Code**: Fixed in 2c7dce9. Replaced the empty string option with an explicit `self-signed` sentinel (default), and the selection step maps it back to an empty `ENCLAVE_CERT_TYPE` so the self-signed path runs unchanged.

File: .github/workflows/e2e-deployment.yml [hunk 2]
```diff
@@ -717,6 +732,11 @@ jobs:
       LZ_OS_VARIANT: ${{ vars.LZ_OS_VARIANT }}
       AAP_LICENSE_FILE: ${{ inputs.aap-license-file }}
       OSAC_CHART_VERSION: ${{ inputs.osac-chart-version }}
+      ENCLAVE_IRONIC_HTTPS: 'true'
+      HETZNER_API_TOKEN: ${{ secrets.HETZNER_API_TOKEN }}
+      ACME_EMAIL: ${{ vars.ACME_EMAIL }}
+      ZEROSSL_EAB_KID: ${{ secrets.ZEROSSL_EAB_KID }}
+      ZEROSSL_EAB_HMAC_KEY: ${{ secrets.ZEROSSL_EAB_HMAC_KEY }}
```
[the remark this answers is not in the document]
Comment (rporres) [reply .github/workflows/e2e-deployment.yml]: ✨ **Claude Code**: Fixed in 2c7dce9. HETZNER_API_TOKEN, ACME_EMAIL, ZEROSSL_EAB_KID and ZEROSSL_EAB_HMAC_KEY are now only present in the step-level `env:` of the two steps that actually invoke cert-gen ("Install Enclave Lab" and "Generate Ironic ISO server TLS certificate"). They no longer appear at the job level.

File: .github/workflows/e2e-deployment.yml [hunk 3]
```diff
@@ -734,6 +754,22 @@ jobs:
...
+          fi
+          if [ "${CERT_TYPE}" = "auto" ]; then
+              TYPES=(le-rsa le-ecdsa-long le-ecdsa-short zerossl-rsa zerossl-ecdsa)
+              CERT_TYPE="${TYPES[$(( ($(date +%u) - 1) % 5 ))]}"
+          fi
+          if [ -n "${CERT_TYPE}" ]; then
+              echo "ENCLAVE_CERT_TYPE=${CERT_TYPE}" >> "$GITHUB_ENV"
+              echo "ENCLAVE_BASE_DOMAIN=nodns.in" >> "$GITHUB_ENV"
+              echo "Certificate type for this run: ${CERT_TYPE}"
+          fi
```
[the remark this answers is not in the document]
Comment (rporres) [reply .github/workflows/e2e-deployment.yml:771]: ✨ **Claude Code**: Fixed in 2c7dce9. `${{ inputs.cert-type }}` is no longer interpolated directly into the run block. It is now passed through an `env:` variable (`INPUT_CERT_TYPE`) and referenced as `${INPUT_CERT_TYPE}` in the shell script. An allowlist case statement additionally rejects any value that is not one of the five known cert types.

File: scripts/deployment/generate_ironic_cert.sh [hunk 4]
```diff
@@ -74,6 +74,29 @@ LZ_BMC_HOSTNAME=$(ssh_exec "awk '/^lzBmcHostname:/ {print \$2}' ${LZ_ENCLAVE_DIR
...
+        {
+            echo "ENCLAVE_IRONIC_CERT<<EOF"
+            echo "${IRONIC_YAML}" | python3 -c \
+                "import sys,yaml; d=yaml.safe_load(sys.stdin); print(d['ironicHTTPSCertificate'].strip())"
+            echo "EOF"
+            echo "ENCLAVE_IRONIC_KEY<<EOF"
+            echo "${IRONIC_YAML}" | python3 -c \
+                "import sys,yaml; d=yaml.safe_load(sys.stdin); print(d['ironicHTTPSKey'].strip())"
+            echo "EOF"
+        } >> "$GITHUB_ENV"
```
[the remark this answers is not in the document]
Comment (rporres) [reply scripts/deployment/generate_ironic_cert.sh]: ✨ **Claude Code**: Fixed in 2c7dce9. Both `python3 -c` calls changed to `uv run python -c`, which uses the project virtualenv where PyYAML is guaranteed to be available.

File: scripts/infrastructure/generate_enclave_vars.sh [hunk 5]
```diff
@@ -72,7 +72,7 @@ LZ_BMC_IP=$(echo "$BMC_CIDR" | sed 's|/.*||' | awk -F. '{print $1"."$2"."$3".2"}
 MASTER_COUNT=$(jq -r '.vms.masters | length' "$ENVIRONMENT_JSON")
 
 # Get base domain from config or use default
-BASE_DOMAIN="${CLUSTER_DOMAIN:-${CLUSTER_NAME}.lab}"
+BASE_DOMAIN="${ENCLAVE_BASE_DOMAIN:-lab}"
```
[the remark this answers is not in the document]
Comment (rporres) [reply scripts/infrastructure/generate_enclave_vars.sh]: ✨ **Claude Code**: Fixed in 2c7dce9. Added `require_env_var "ENCLAVE_BASE_DOMAIN"` inside the `ENCLAVE_CERT_TYPE` branch in both `generate_enclave_vars.sh` and `generate_ironic_cert.sh`, so missing the variable fails fast with a clear error before any certbot invocation.

File: scripts/infrastructure/generate_enclave_vars.sh [hunk 6]
```diff
@@ -213,42 +213,52 @@ for i in $(seq 0 $((MASTER_COUNT - 1))); do
...
-cat > "$CERTS_VARS_OUTPUT" <<EOF
+
+if [ -n "${ENCLAVE_CERT_TYPE:-}" ]; then
+    info "Requesting real TLS certificate (${ENCLAVE_CERT_TYPE}) via enclave-cert-gen..."
+    uv run --group cert-gen enclave-cert-gen ingress-api \
+        --san "api.${CLUSTER_NAME}.${BASE_DOMAIN}" \
+        --san "*.apps.${CLUSTER_NAME}.${BASE_DOMAIN}" \
+        --type "${ENCLAVE_CERT_TYPE}" \
+        > "$CERTS_VARS_OUTPUT"
+    info "✓ Real CA certificates written for api.${CLUSTER_FQDN} and *.apps.${CLUSTER_FQDN}"
```
Comment (rporres) [inline scripts/infrastructure/generate_enclave_vars.sh:229]: ✨ **Claude Code**: Fixed in 2c7dce9. The real-CA output is now written to a `mktemp` file, `chmod 600`-ed, then atomically moved into `$CERTS_VARS_OUTPUT`. This prevents a truncated destination on failure and ensures the private key material is mode 600 from the moment it lands.

[the remark this answers is not in the document]
Comment (rporres) [reply scripts/infrastructure/generate_enclave_vars.sh:229]: ✨ **Claude Code**: Fixed in 2c7dce9. The summary line is now conditional: when `ENCLAVE_CERT_TYPE` is set it prints "Real CA (${ENCLAVE_CERT_TYPE}) via enclave-cert-gen"; otherwise it still prints "Self-signed (generated for CI/testing)".

File: scripts/setup/configure_devscripts.sh [hunk 7]
```diff
@@ -185,11 +185,11 @@ export MANAGE_INT_BRIDGE="n"
...
 
-# Cluster domain (for DNS)
-export CLUSTER_DOMAIN="${ENCLAVE_CLUSTER_NAME}.lab"
-
 # Base domain (so dev-scripts uses .lab not test.metalkube.org for CLUSTER_DOMAIN)
-export BASE_DOMAIN="lab"
+export BASE_DOMAIN="${ENCLAVE_BASE_DOMAIN:-lab}"
+
+# Cluster domain (for DNS)
+export CLUSTER_DOMAIN="${ENCLAVE_CLUSTER_NAME}.${BASE_DOMAIN}"
```
[the remark this answers is not in the document]
Comment (rporres) [reply scripts/setup/configure_devscripts.sh]: ✨ **Claude Code**: Fixed in 2c7dce9. Escaped `\${BASE_DOMAIN}` in the heredoc so it is written literally into the generated file and expanded at source-time (when BASE_DOMAIN is already set), not at heredoc-generation time when the variable is unset.

File: src/enclave/cert_gen/main.py [hunk 8]
```diff
@@ -0,0 +1,348 @@
...
+    for domain in domains:
+        args += ["-d", domain]
+
+    result = subprocess.run(args, capture_output=True, text=True, check=False)
+    if result.returncode != 0:
+        output = result.stdout + result.stderr
+        match = re.search(r"retry after \S+", output, re.IGNORECASE)
+        if match:
+            raise RuntimeError(f"CA rate limit reached — {match.group()}")
+        raise RuntimeError(f"Certificate issuance failed:\n{output}")
```
[the remark this answers is not in the document]
Comment (rporres) [reply src/enclave/cert_gen/main.py]: ✨ **Claude Code**: Fixed in 2c7dce9. Before constructing the exception message, `eab_kid` and `eab_hmac_key` are replaced with `***REDACTED***` in the combined stdout/stderr string. The credentials can still appear in certbot's argv (visible via `ps`), but they will not reach exception messages, logs, or CI step summaries.

[the remark this answers is not in the document]
Comment (rporres) [reply src/enclave/cert_gen/main.py]: ✨ **Claude Code**: Fixed in 2c7dce9. Added `CertIssuanceError` (replaces `RuntimeError` in `issue_cert`) and `RootCaExtractionError` (replaces `ValueError` in `_aia_issuer_url`, `_fetch_cert`, and `extract_root_ca`). All callers and tests updated to use the new types.

File: src/enclave/cert_gen/main.py [hunk 9]
```diff
@@ -0,0 +1,348 @@
...
+            return desc.access_location.value
+    raise ValueError(f"No CA Issuers URI in AIA of cert: {cert.subject}")
+
+
+def _fetch_cert(url: str) -> x509.Certificate:
+    data = requests.get(url, timeout=15).content
+    try:
+        return x509.load_der_x509_certificate(data)
+    except ValueError:
+        return x509.load_pem_x509_certificate(data)
```
[the remark this answers is not in the document]
Comment (rporres) [reply src/enclave/cert_gen/main.py]: ✨ **Claude Code**: Fixed in 2c7dce9. `_fetch_cert` now: (1) enforces an http/https scheme allowlist before making any request, (2) caps the consumed response body at 64 KiB via `resp.content[:65536]`, and (3) wraps `requests.RequestException` into `RootCaExtractionError` so parse failures and network errors all surface through the same exception type.

File: src/enclave/cert_gen/main.py [hunk 10]
```diff
@@ -0,0 +1,348 @@
...
+def cmd_ingress_api(sans: tuple[str, ...], cert_type: str, root_ca: bool) -> None:
+    """Issue a multi-SAN cert and print certificates.yaml fields to stdout."""
+    token = _get_env("HETZNER_API_TOKEN")
+    email = _get_env("ACME_EMAIL")
+    eab_kid = os.environ.get("ZEROSSL_EAB_KID", "")
+    eab_hmac_key = os.environ.get("ZEROSSL_EAB_HMAC_KEY", "")
+    if cert_type.startswith("zerossl") and not (eab_kid and eab_hmac_key):
+        raise click.UsageError(
+            "ZEROSSL_EAB_KID and ZEROSSL_EAB_HMAC_KEY are required for zerossl cert types"
+        )
```
[the remark this answers is not in the document]
Comment (rporres) [reply src/enclave/cert_gen/main.py]: ✨ **Claude Code**: Fixed in 2c7dce9. The `UsageError` string is now split across two lines so each stays within the 88-character Ruff limit.

File: src/tests/test_cert_gen.py [hunk 11]
```diff
@@ -0,0 +1,486 @@
...
+    def test_zerossl_missing_eab(self) -> None:
+        runner = CliRunner()
+        result = runner.invoke(
+            cli,
+            ["ingress-api", "--san", "api.x.nodns.in", "--type", "zerossl-rsa"],
+            env=self._env(),
+            catch_exceptions=False,
+        )
+        assert result.exit_code != 0
+        assert "ZEROSSL_EAB_KID" in result.output
```
[the remark this answers is not in the document]
Comment (rporres) [reply src/tests/test_cert_gen.py:505]: ✨ **Claude Code**: Fixed in 2c7dce9. The missing-token tests now pass `env={"HETZNER_API_TOKEN": None}` (and `"ACME_EMAIL": None` where appropriate) to CliRunner, which strips those keys from the ambient environment so the test is not sensitive to locally exported variables.

**Comment by rporres** [comment]
@coderabbitai approve

File: scripts/infrastructure/generate_enclave_vars.sh [hunk 12]
```diff
@@ -213,42 +213,56 @@ for i in $(seq 0 $((MASTER_COUNT - 1))); do
...
+    info "Requesting real TLS certificate (${ENCLAVE_CERT_TYPE}) via enclave-cert-gen..."
+    CERTS_VARS_TEMP=$(mktemp)
+    uv run --group cert-gen enclave-cert-gen ingress-api \
+        --san "api.${CLUSTER_NAME}.${BASE_DOMAIN}" \
+        --san "*.apps.${CLUSTER_NAME}.${BASE_DOMAIN}" \
+        --type "${ENCLAVE_CERT_TYPE}" \
+        > "${CERTS_VARS_TEMP}"
+    chmod 600 "${CERTS_VARS_TEMP}"
+    mv "${CERTS_VARS_TEMP}" "${CERTS_VARS_OUTPUT}"
+    info "✓ Real CA certificates written for api.${CLUSTER_FQDN} and *.apps.${CLUSTER_FQDN}"
```
[the remark this answers is not in the document]
Comment (rporres) [reply scripts/infrastructure/generate_enclave_vars.sh:229]: ✨ **Claude Code**: Fixed in 9aab854. The temp file is now created with `mktemp "$(dirname "$CERTS_VARS_OUTPUT")/certificates.XXXXXX"` so it lands on the same filesystem as the destination, keeping `mv` atomic. A `trap 'rm -f "$CERTS_VARS_TEMP"' EXIT` is set immediately after creation and cleared with `trap - EXIT` after the successful `mv`, so any failure between creation and rename cleans up the partial file.

File: src/enclave/cert_gen/main.py [hunk 13]
```diff
@@ -0,0 +1,416 @@
...
+    except requests.RequestException as exc:
+        raise RootCaExtractionError(f"Failed to fetch AIA issuer {url}: {exc}") from exc
+    data = resp.content[:65536]
+    try:
+        return x509.load_der_x509_certificate(data)
+    except ValueError:
+        try:
+            return x509.load_pem_x509_certificate(data)
+        except ValueError as exc:
+            raise RootCaExtractionError(f"Cannot parse cert from {url}") from exc
```
[the remark this answers is not in the document]
Comment (rporres) [reply src/enclave/cert_gen/main.py:225]: ✨ **Claude Code**: Fixed in 9aab854. `_fetch_cert` now uses `requests.get(..., stream=True)` inside a `with` block (which closes the connection on exit) and reads at most 64 KiB from the socket via `resp.raw.read(65536, decode_content=True)`. The response body is never fully buffered regardless of its size. The test updated accordingly: `mock_resp.raw.read.return_value = root_der` with explicit context-manager hooks.

File: validations.sh [hunk 14]
```diff
@@ -71,8 +71,9 @@ checkDNS(){
         validation fail $check_name "Failed to resolve $name"
     fi
 
-    if [[ $result != $value ]]; then
-        validation fail $check_name "$name points to $result. It does not match with $value"
+    if ! echo "$result" | grep -qx "$value"; then
```
[the remark this answers is not in the document]
Comment (rporres) [reply validations.sh]: ✨ **Claude Code**: Fixed in 9aab854. Changed to `grep -Fqx -- "$value"`: `-F` treats the expected IP as a fixed string (no regex interpretation), and `--` prevents values that start with `-` from being parsed as flags.

File: scripts/deployment/generate_ironic_cert.sh [hunk 15]
```diff
@@ -74,6 +74,30 @@ LZ_BMC_HOSTNAME=$(ssh_exec "awk '/^lzBmcHostname:/ {print \$2}' ${LZ_ENCLAVE_DIR
 
 info "Generating TLS certificate for Ironic ISO server"
 
+if [ -n "${ENCLAVE_CERT_TYPE:-}" ]; then
+    require_env_var "ENCLAVE_BASE_DOMAIN"
+    info "Requesting real ironic TLS certificate (${ENCLAVE_CERT_TYPE}) via enclave-cert-gen..."
+    IRONIC_YAML=$(uv run --group cert-gen enclave-cert-gen ironic \
+        --san "ironic.${ENCLAVE_CLUSTER_NAME}.${ENCLAVE_BASE_DOMAIN}" \
+        --type "${ENCLAVE_CERT_TYPE}")
```
[the remark this answers is not in the document]
Comment (rporres) [reply scripts/deployment/generate_ironic_cert.sh]: ✨ **Claude Code**: Fixed in 826178c. Moved the real-CA block to before the CA file check and SSH setup. The block now exits early when `ENCLAVE_CERT_TYPE` is set without requiring SSH connectivity or local CA files. Added a note in the script header that public CAs (Let's Encrypt, ZeroSSL) cannot issue IP SANs and that `lzBmcHostname` must be configured in `config/global.yaml` for TLS verification to succeed.

File: scripts/deployment/generate_ironic_cert.sh [hunk 16]
```diff
@@ -74,6 +74,30 @@ LZ_BMC_HOSTNAME=$(ssh_exec "awk '/^lzBmcHostname:/ {print \$2}' ${LZ_ENCLAVE_DIR
...
+        {
+            echo "ENCLAVE_IRONIC_CERT<<EOF"
+            echo "${IRONIC_YAML}" | uv run python -c \
+                "import sys,yaml; d=yaml.safe_load(sys.stdin); print(d['ironicHTTPSCertificate'].strip())"
+            echo "EOF"
+            echo "ENCLAVE_IRONIC_KEY<<EOF"
+            echo "${IRONIC_YAML}" | uv run python -c \
+                "import sys,yaml; d=yaml.safe_load(sys.stdin); print(d['ironicHTTPSKey'].strip())"
+            echo "EOF"
+        } >> "$GITHUB_ENV"
```
[the remark this answers is not in the document]
Comment (rporres) [reply scripts/deployment/generate_ironic_cert.sh]: ✨ **Claude Code**: Fixed in 826178c. Both `ironicHTTPSCertificate` and `ironicHTTPSKey` are now parsed into shell variables (`IRONIC_CERT`, `IRONIC_KEY`) before the `{...} >> "$GITHUB_ENV"` write group is opened. A YAML parse failure now aborts via `set -e` before any write to `GITHUB_ENV`, leaving the file intact.

File: src/enclave/cert_gen/main.py [hunk 17]
```diff
@@ -0,0 +1,416 @@
...
+        "--force-renewal",
+    ]
+    if key_type == "rsa":
+        args += ["--rsa-key-size", key_param]
+    else:
+        args += ["--elliptic-curve", key_param]
+    if preferred_chain:
+        args += ["--preferred-chain", preferred_chain]
+    if eab_kid and eab_hmac_key:
+        args += ["--eab-kid", eab_kid, "--eab-hmac-key", eab_hmac_key]
```
[the remark this answers is not in the document]
Comment (rporres) [reply src/enclave/cert_gen/main.py]: ✨ **Claude Code**: Fixed in 826178c. EAB credentials are now written to a `0600` file (`config_dir/eab.ini`) and passed to certbot via `--config <path>` instead of `--eab-kid`/`--eab-hmac-key`. The credentials never appear in the process argument list. The existing EAB redaction in the error message is kept as a defence-in-depth measure.

File: src/enclave/cert_gen/main.py [hunk 18]
```diff
@@ -0,0 +1,416 @@
...
+def _get_env(name: str) -> str:
+    val = os.environ.get(name, "")
+    if not val:
+        raise click.UsageError(f"Missing required environment variable: {name}")
+    return val
+
+
+@click.group()
+def cli() -> None:
+    """Issue real CA TLS certificates for enclave deployments."""
```
[the remark this answers is not in the document]
Comment (rporres) [reply src/enclave/cert_gen/main.py:346]: ✨ **Claude Code**: Not addressing in this PR. `enclave-cert-gen` is invoked programmatically (via `uv run`); it has no `--log-level` or `--log-format` CLI options, and `configure_logging` has nothing to bind to. Adding those options would change the CLI interface and is a separate feature beyond the scope of this PR.

File: src/enclave/cert_gen/main.py [hunk 19]
```diff
@@ -0,0 +1,416 @@
...
+    """Issue a multi-SAN cert and print certificates.yaml fields to stdout."""
+    token = _get_env("HETZNER_API_TOKEN")
+    email = _get_env("ACME_EMAIL")
+    eab_kid = os.environ.get("ZEROSSL_EAB_KID", "")
+    eab_hmac_key = os.environ.get("ZEROSSL_EAB_HMAC_KEY", "")
+    if cert_type.startswith("zerossl") and not (eab_kid and eab_hmac_key):
+        raise click.UsageError(
+            "ZEROSSL_EAB_KID and ZEROSSL_EAB_HMAC_KEY are required "
+            "for zerossl cert types"
+        )
```
[the remark this answers is not in the document]
Comment (rporres) [reply src/enclave/cert_gen/main.py]: ✨ **Claude Code**: Fixed in 826178c. Extracted `_load_credentials(cert_type) -> tuple[str, str, str, str]` which returns `(token, email, eab_kid, eab_hmac_key)` and handles the ZeroSSL EAB validation. Both `cmd_ingress_api` and `cmd_ironic` now call it in one line, eliminating the duplicated block.

File: src/tests/test_cert_gen.py [hunk 20]
```diff
@@ -0,0 +1,491 @@
...
+        mock_resp = MagicMock()
+        mock_resp.__enter__ = lambda s: s
+        mock_resp.__exit__ = MagicMock(return_value=False)
+        mock_resp.raw.read.return_value = root_der
+
+        with patch("enclave.cert_gen.main.requests.get", return_value=mock_resp):
+            result = extract_root_ca(chain_path)
+
+        cert = x509.load_pem_x509_certificate(result.encode())
+        assert cert.subject == cert.issuer
```
[the remark this answers is not in the document]
Comment (rporres) [reply src/tests/test_cert_gen.py:186]: ✨ **Claude Code**: Fixed in 826178c. Added: `TestFetchCert` with three cases (invalid scheme, `RequestException`, unparsable body); `test_aia_walk_exhausted_after_five_hops` verifying the loop raises after exactly 5 non-root hops; `TestLoadCredentials` covering missing token, missing ZeroSSL EAB, and successful LE credential load; and updated `test_eab_credentials_included` to assert `--eab-kid` is absent from argv, that a `0600` config file is created with the correct content, and that `--config` is passed instead.

**Comment by rporres** [comment]
@coderabbitai approve

File: operators/quay-operator/clair_validations.yaml [hunk 21]
```diff
@@ -11,24 +24,119 @@
...
+        api_version: apps/v1
+        kind: Deployment
+        name: registry-clair-app
+        namespace: quay-enterprise
+      register: debug_clair_deploy
+      failed_when: false
+
+    - name: "[debug] Print Clair deployment status"
+      ansible.builtin.debug:
+        var: debug_clair_deploy.resources[0].status
```
[the remark this answers is not in the document]
Comment (rporres) [reply operators/quay-operator/clair_validations.yaml]: removed from the PR, it was a debugging code for a test

File: operators/quay-operator/clair_validations.yaml [hunk 22]
```diff
@@ -11,24 +24,119 @@
...
+      kubernetes.core.k8s_exec:
+        namespace: quay-enterprise
+        pod: "{{ clair_pod_name }}"
+        command: ls -la /data/
+      register: debug_data_dir
+      failed_when: false
+
+    - name: "[debug] Print /data directory listing"
+      ansible.builtin.debug:
+        var: debug_data_dir
```
[the remark this answers is not in the document]
Comment (rporres) [reply operators/quay-operator/clair_validations.yaml]: removed from the PR, it was a debugging code for a test

File: operators/quay-operator/clair_validations.yaml [hunk 23]
```diff
@@ -11,24 +24,119 @@
...
+    - name: "[debug] Get quay-enterprise namespace events"
+      kubernetes.core.k8s_info:
+        kind: Event
+        namespace: quay-enterprise
+      register: debug_ns_events
+      failed_when: false
+
+    - name: "[debug] Print Clair-related events"
+      ansible.builtin.debug:
+        msg: "{{ debug_ns_events.resources | default([]) | selectattr('involvedObject.name', 'match', 'registry-clair') | list }}"
```
[the remark this answers is not in the document]
Comment (rporres) [reply operators/quay-operator/clair_validations.yaml]: removed from the PR, it was a debugging code for a test

File: operators/quay-operator/clair_validations.yaml [hunk 24]
```diff
@@ -1,34 +1,196 @@
...
+         | map(attribute='status.containerStatuses')
+         | map('map', attribute='restartCount')
+         | map('list')
+         | map('min')
+         | list }}
+    min_restarts: "{{ restart_counts | min }}"
+    min_index: "{{ restart_counts.index(min_restarts | int) }}"
   ansible.builtin.set_fact:
-    clair_pod_name: "{{ clair_pods.resources[0].metadata.name }}"
+    clair_pod_name: "{{ clair_pods.resources[min_index | int].metadata.name }}"
```
[the remark this answers is not in the document]
Comment (rporres) [reply operators/quay-operator/clair_validations.yaml]: removed from the PR, it was a debugging code for a test

File: operators/quay-operator/clair_validations.yaml [hunk 25]
```diff
@@ -1,34 +1,196 @@
...
+          - quay-enterprise
+          - "{{ item.metadata.name }}"
+          - --all-containers=true
+          - --previous
+          - --tail=50
+      loop: "{{ debug_clair_pods.resources | default([]) }}"
+      register: debug_oc_logs_prev
+      failed_when: false
+      environment:
+        KUBECONFIG: "{{ workingDir }}/ocp-cluster/auth/kubeconfig"
```
[the remark this answers is not in the document]
Comment (rporres) [reply operators/quay-operator/clair_validations.yaml]: removed from the PR, it was a debugging code for a test

File: scripts/deployment/generate_ironic_cert.sh [hunk 26]
```diff
@@ -2,8 +2,14 @@
 # Generate a TLS certificate for the Ironic ISO server, signed by the CA
 # produced by generate_ironic_ca.sh
 #
-# Reads lzBmcIP from config/global.yaml on the Landing Zone to set the
-# certificate SAN. Key and CSR are generated locally on the CI runner and
+# When ENCLAVE_CERT_TYPE is set, requests a real certificate from a public CA
+# via enclave-cert-gen and exits early without needing SSH or local CA files.
+# Note: public CAs (Let's Encrypt, ZeroSSL) cannot issue IP SANs; the SAN must
+# be a DNS name. Set lzBmcHostname in config/global.yaml so that TLS validation
+# can resolve the cert via hostname rather than IP.
```
[the remark this answers is not in the document]
Comment (rporres) [reply scripts/deployment/generate_ironic_cert.sh]: fixed in 499d1dd

File: src/tests/test_cert_gen.py [hunk 27]
```diff
@@ -0,0 +1,588 @@
...
+        assert "--eab-kid" not in args
+        assert "kid123" not in args
+        # Instead they are written to a 0600 config file passed via --config.
+        eab_ini = tmp_path / "eab.ini"
+        assert eab_ini.exists()
+        content = eab_ini.read_text(encoding="utf-8")
+        assert "eab-kid = kid123" in content
+        assert "eab-hmac-key = hmac456" in content
+        assert "--config" in args
+        assert oct(eab_ini.stat().st_mode)[-3:] == "600"
```
[the remark this answers is not in the document]
Comment (rporres) [reply src/tests/test_cert_gen.py:401]: fixed in 499d1dd

**Comment by rporres** [comment]
✨ **Claude Code**: Tracked in [OSAC-4486](https://redhat.atlassian.net/browse/OSAC-4486)

**Comment by rporres** [comment]
@coderabbitai please approve if there are no outstanding issues

**Review by maorfr — APPROVED** [review]

**Review by eurijon — APPROVED** [review]
